kernel/src/table_changes/scan.rs

```diff
@@ -0,0 +1,204 @@
+use std::sync::Arc;
+
+use itertools::Itertools;
+use tracing::debug;
+
+use crate::scan::ColumnType;
+use crate::schema::SchemaRef;
+use crate::{DeltaResult, ExpressionRef};
+
+use super::{TableChanges, CDF_FIELDS};
+
+/// The result of building a [`TableChanges`] scan over a table. This can be used to get a change
+/// data feed from the table
+#[derive(Debug)]
+#[allow(unused)]
+pub struct TableChangesScan {
+    table_changes: Arc<TableChanges>,
+    logical_schema: SchemaRef,
+    predicate: Option<ExpressionRef>,
+    all_fields: Vec<ColumnType>,
+    have_partition_cols: bool,
+}
+
+/// Builder to read the `TableChanges` of a table.
+pub struct TableChangesScanBuilder {
```

Comment:
maybe flag that this shares a lot with normal scan builder?

Reply:
Done. lmk if you think it needs an issue or more details.